docs/nativetypes.rst

```diff
@@ -0,0 +1,28 @@
+Native Python Types
+===================
+
+The Jinja2 :class:`NativeEnvironment` class can be used instead of :class:`Environment` to override jinja's default behavior of returning only strings. This can be useful if you are using jinja outside the context of creating text files.
```

Comment:
Needs to describe what the use case is in more detail. A good start:

> The default `Environment` renders to a string. With `NativeEnvironment`, the result of `render` is a Python type if the result is a single node or can be interpreted as a Python literal. This is useful, for example in Ansible (link to Ansible docs about this), as an intermediate step where the user can use templates to generate variables, which are later passed to a regular template.

Describe the problem that this solves so that users know if they should use it.